### ISSUE TITLE
Use opentelemetry-lib for OTel enrichments

### ISSUE THREAD

**Issue by lahsivjar** [elastic/apm-data#387]
https://github.com/elastic/opentelemetry-lib/tree/main/enrichments/trace implements enrichment of OTel datamodel with attributes required for APM - similar to what we do [in APM data](https://github.com/elastic/apm-data/blob/main/input/otlp/traces.go). The opentelemetry-lib implementation removes dependency from some of the old semconv and Jaeger which could help us to eliminate some of the tech debt. The enrichment currently only works for traces but opentelemetry-lib could be easily extended to metrics and logs.

In addition to enrichments, we would also need translation of OTel attributes to Elastic APM attributes in opentelemetry-lib. These mappings are already [captured in otel-data plugin](x-pack/plugin/otel-data/src/main/resources/component-templates/semconv-resource-to-ecs@mappings.yaml) and would need to be coded, probably in the opentelemetry-lib itself.

To bring the changes to apm-data, the code flow would need to be modified. Instead of translating the OTel data model to APM data model and applying enrichments, we would apply enrichment and then translate to APM data model.

Note that the changes done here will also impact MIS.

WIP: document exact differences between APM data handling and opentelemetry-lib handling

**Comment by raultorrecilla** [comment]
moving this from it105 to it106 as it's marked for 9.0, is this correct @lahsivjar ?

**Comment by lahsivjar** [comment]
> moving this from it105 to it106 as it's marked for 9.0, is this correct @lahsivjar ?

Yes, this sounds right. Introducing opentelemetry-lib would be breaking in some sense as the library drops some of the handling or deprecated attributes done in apm-data.

**Comment by simitt** [comment]
@lahsivjar and I discussed this topic further. We are not allowed to introduce breaking changes in MIS, which also depends on `apm-data`. Given the focus on MOTel and moving away from apm-server and eventually also prefering MOTel over MIS, we are not moving forward with using opentelemetry-lib for OTel enrichments at this point.
